### PR TITLE
Removed travis-5.2.x.cfg and updated docs

### PR DESCRIPTION
**Pull request by mauritsvanrees**
The documentation no longer advertises to use the `buildout-cache.cfg` anymore, which is what happens with the travis configs.
Plus other small fixes in the documentation.
This fixes issue #51.

Additionally we could try to warn current users of the `travis-*.cfg` files by making sure an extra [deprecated part](https://github.com/collective/buildout.plonetest/pull/50#issuecomment-523032360) gets loaded. But we can try that afterwards.

First let's have better documentation.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -6,40 +6,56 @@ Testing/development buildouts for Plone
...
     python: 2.7
+    sudo: false
+    cache:
+      pip: true
+      directories:
+      - eggs
+    before_install:
+    - virtualenv -p `which python` .
+    - bin/pip install -r requirements.txt
+    - bin/buildout -N -t 3 annotate
```
Comment (idgserpro) [inline README.rst]: I think the indentation here would be good.

File: README.rst [hunk 2]
```diff
@@ -6,40 +6,56 @@ Testing/development buildouts for Plone
...
     install:
-      - mkdir -p buildout-cache/downloads
-      - python bootstrap.py -c travis.cfg
-      - bin/buildout -N -t 3 -c travis.cfg
-    script: bin/test
+    - bin/buildout -N -t 3
```
Comment (idgserpro) [inline README.rst]: I think the indentation here would be good.

I really don't like of the `-N`. It freezes the unpinned package version. So we lost integration with new versions. But this is out of scope.

File: README.rst [hunk 3]
```diff
@@ -6,40 +6,56 @@ Testing/development buildouts for Plone
...
+    script:
+    - bin/test
```
Comment (idgserpro) [inline README.rst]: I think the indentation here would be good.

File: README.rst [hunk 4]
```diff
@@ -48,21 +64,25 @@ Include the following in your buildout configuration to create an `i18ndude`_
...
+After running ``bin/buildout`` you will find a ``bin/rebuild_i18n.sh``; run the
 script and the po files will be updated.
 
-Domain name is taken from the ${buildout:package-name} variable; Plone domain
-is also included.
+Domain name is taken from the ``${buildout:package-name}`` variable.
+The plone domain is also included.
+
+.. ATTENTION::
+   Originally, the ``rebuild_i18n-sh`` part used a `filesystem template <https://github.com/collective/buildout.plonetest/blob/master/templates/rebuild_i18n.sh.in>`_.
```
Comment (idgserpro) [inline README.rst]: I think this file could be removed.

File: README.rst [hunk 5]
```diff
@@ -73,45 +93,77 @@ Update your ``.travis.yml`` file with the following::
...
-    allow-hosts +=
-        code.google.com
-        robotframework.googlecode.com
+    [versions]
+    setuptools = 41.2.0
+    zc.buildout = 2.13.2
+
+The versions are optional, but they can help in avoiding restarts when buildout tries to upgrade itself to a version pinned in Plone.
+It is fine to start without them, but when you run into problems in `Travis CI`_, consider adding them.
+In this way, all Plone versions use the same buildout and setuptools versions.
```
Comment (idgserpro) [inline README.rst]: Shouldn't each version of Plone have its own version of zc.buildout/setuptools?

see:

https://github.com/plone/bobtemplates.plone/blob/4a792eb0899317e406310661eee5b55d40a5d915/bobtemplates/plone/addon/.travis.yml.bob#L25

File: README.rst [hunk 6]
```diff
@@ -73,45 +93,77 @@ Update your ``.travis.yml`` file with the following::
       - export DISPLAY=:99.0
       - sh -e /etc/init.d/xvfb start
```
Comment (idgserpro) [inline README.rst]: I think it doesn't work anymore.

see:

https://github.com/collective/collective.cover/pull/866
https://docs.travis-ci.com/user/gui-and-headless-browsers/#using-xvfb-to-run-tests-that-require-a-gui

File: README.rst [hunk 7]
```diff
@@ -73,45 +93,77 @@ Update your ``.travis.yml`` file with the following::
...
+          env: PLONE_VERSION="5.2"
+        - python: "3.7"
+          env: PLONE_VERSION="5.2"
+      sudo: true
+      fast_finish: true
+    before_install:
+    - virtualenv -p `which python` .
+    - bin/pip install -r requirements.txt
+    - sed -ie "s#test-5.x.cfg#test-$PLONE_VERSION.x.cfg#" buildout.cfg
+    - bin/buildout -N -t 3 annotate
```
Comment (idgserpro) [inline README.rst]: I think the indentation here would be good.

File: README.rst [hunk 8]
```diff
@@ -73,45 +93,77 @@ Update your ``.travis.yml`` file with the following::
...
     install:
-      - sed -ie "s#travis-4.x.cfg#travis-$PLONE_VERSION.x.cfg#" travis.cfg
-      - mkdir -p buildout-cache/downloads
-      - python bootstrap.py -c travis.cfg
-      - bin/buildout -c travis.cfg -N -q -t 3
-    script: bin/test
+    - bin/buildout -N -t 3
```
Comment (idgserpro) [inline README.rst]: I think the indentation here would be good.

File: README.rst [hunk 9]
```diff
@@ -73,45 +93,77 @@ Update your ``.travis.yml`` file with the following::
...
+    script:
+    - bin/test
```
Comment (idgserpro) [inline README.rst]: I think the indentation here would be good.

File: README.rst [hunk 10]
```diff
@@ -130,23 +182,29 @@ update your ``travis.cfg`` file like::
...
 
 and update your ``.travis.yml`` like::
 
     language: python
     python: 2.7
+    sudo: false
+    cache:
+      pip: true
+      directories:
+      - eggs
```
Comment (idgserpro) [inline README.rst]: I think the indentation here would be good.

File: README.rst [hunk 11]
```diff
@@ -130,23 +182,29 @@ update your ``travis.cfg`` file like::
...
-    install: 
-      - mkdir -p buildout-cache/downloads
-      - python bootstrap.py -c travis.cfg
-      - bin/buildout -c travis.cfg -N -q -t 3
-    
-    script: bin/$TARGET
+    before_install:
+    - virtualenv -p `which python` .
+    - bin/pip install -r requirements.txt
+    - bin/buildout -N -t 3 annotate
```
Comment (idgserpro) [inline README.rst]: I think the indentation here would be good.

File: README.rst [hunk 12]
```diff
@@ -130,23 +182,29 @@ update your ``travis.cfg`` file like::
...
+    install:
+    - bin/buildout -N -t 3
+    script:
+    - bin/$TARGET
```
Comment (idgserpro) [inline README.rst]: I think the indentation here would be good.

File: README.rst [hunk 13]
```diff
@@ -130,23 +182,29 @@ update your ``travis.cfg`` file like::
...
-      - python bootstrap.py -c travis.cfg
-      - bin/buildout -c travis.cfg -N -q -t 3
-    
-    script: bin/$TARGET
+    before_install:
+    - virtualenv -p `which python` .
+    - bin/pip install -r requirements.txt
+    - bin/buildout -N -t 3 annotate
+    install:
+    - bin/buildout -N -t 3
```
Comment (idgserpro) [inline README.rst]: I think the indentation here would be good.

**Comment by idgserpro** [comment]
@mauritsvanrees since `travis-5.x.cfg` is broken too, I think it might be empty, only with a deprecated warning.

File: README.rst (hunk 1 above)
Comment (htgoebel) [inline README.rst]: This last line shouldn't be here, should it? It's duplicate with the one in the `install` section just below.

Comment (mauritsvanrees) [reply README.rst]: I see both with and without indentation used in various projects. Indentation *does* give a bit of extra clarity. I will do that.

File: README.rst (hunk 2 above)
Comment (mauritsvanrees) [reply README.rst]: For me, the `-N` here is about not upgrading or downgrading `setuptools` or `zc.buildout`, though it also affects other versions. Depending on what is pinned in the versions, and which versions of setuptools and zc.buildout are already installed at this point, I have too often seen `bin/buildout` go wrong, ending up in a loop of upgrading those two packages, where a `bin/buildout -N` would have worked fine.

But yes, out of scope to change this. And we could add more comments here with advice about this, but that probably just complicates the readme, confusing people more than it helps.

File: README.rst (hunk 4 above)
Comment (mauritsvanrees) [reply README.rst]: Eventually the file could be removed, but current master is still pointing to it, so I think we should leave it in for now.

File: README.rst (hunk 5 above)
Comment (mauritsvanrees) [reply README.rst]: They can, but they don't have to. Latest buildout and setuptools work fine for all current versions of Plone. With the same version, we avoid the need for extra files.

File: README.rst (hunk 1 above)
Comment (mauritsvanrees) [reply README.rst]: It is not the same. In `before_install` we call `bin/buildout annotate`, so the parsed configuration is printed for debug purposes, and in `install` it is the standard `bin/buildout` call.

**Comment by mauritsvanrees** [comment]
> since travis-5.x.cfg is broken too, I think it might be empty, only with a deprecated warning.

Ah, right, you mentioned that earlier. It uses `plone-series = 5.1` and extends the `5-latest` versions, and that is not a working combination, and cannot easily be fixed.
I will add a comment in the travis file.

**Comment by mauritsvanrees** [comment]
Approved. I merge the PR myself.
